# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 6770fb6b8d92
# Full Hash: 6770fb6b8d9217b7d75ed7169cb1389c868498ed
# Author: Martin Stransky <stransky@redhat.com>
# Date: 2024-09-06 21:37:32
# Regressor Bug: 1915596
# File Overlap Count: 2
# Description:
#   Bug 1915596 [Wayland] Get touchpad hold event directly from Wayland display r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D220551
# ==============================================================================

diff -r 791b8e439bbd -r 6770fb6b8d92 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Fri Sep 06 08:37:00 2024 +0000
+++ b/widget/gtk/nsWindow.cpp	Fri Sep 06 09:16:28 2024 +0000
@@ -141,7 +141,6 @@
 #define MAX_RECTS_IN_REGION 100
 
 #if !GTK_CHECK_VERSION(3, 18, 0)
-
 struct _GdkEventTouchpadPinch {
   GdkEventType type;
   GdkWindow* window;
@@ -159,16 +158,8 @@
   guint state;
 };
 
-typedef enum {
-  GDK_TOUCHPAD_GESTURE_PHASE_BEGIN,
-  GDK_TOUCHPAD_GESTURE_PHASE_UPDATE,
-  GDK_TOUCHPAD_GESTURE_PHASE_END,
-  GDK_TOUCHPAD_GESTURE_PHASE_CANCEL
-} GdkTouchpadGesturePhase;
-
 gint GDK_TOUCHPAD_GESTURE_MASK = 1 << 24;
 GdkEventType GDK_TOUCHPAD_PINCH = static_cast<GdkEventType>(42);
-
 #endif
 
 const gint kEvents = GDK_TOUCHPAD_GESTURE_MASK | GDK_EXPOSURE_MASK |
@@ -5511,6 +5502,11 @@
   return TRUE;
 }
 
+void nsWindow::OnTouchpadHoldEvent(GdkTouchpadGesturePhase aPhase, guint aTime,
+                                   uint32_t aFingers) {
+  LOG("OnTouchpadHoldEvent: aPhase %d aFingers %d", aPhase, aFingers);
+}
+
 gboolean nsWindow::OnTouchEvent(GdkEventTouch* aEvent) {
   LOG("OnTouchEvent: x=%f y=%f type=%d\n", aEvent->x, aEvent->y, aEvent->type);
   if (!mHandleTouchEvent) {